servant-server/src/Servant/Server/Internal.hs

```diff
@@ -821,7 +823,7 @@ instance (HasContextEntry context (NamedContext name subContext), HasServer subA
 -- Erroring instance for 'HasServer' when a combinator is not fully applied
 instance TypeError (PartialApplication HasServer arr) => HasServer ((arr :: a -> b) :> sub) context
   where
-    type ServerT (arr :> sub) _ = TypeError (PartialApplication HasServer arr)
+    type ServerT (arr :> sub) _ = TypeError (PartialApplication (HasServer :: Type -> [Type] -> Constraint) arr)
```

Comment:
Can we get a quick patch release with just this change included? With this I think GHC 9.2.1 compilation will succeed if allow-newer is set and fixed versions of cryptonite, memory, and basement are included in the cabal.project.

Reply:
It was already merged as part of

* haskell-servant/servant/pull/1525